frontend/apps/web/components/onboarding-checklist/WelcomeDialog.tsx

```diff
@@ -0,0 +1,161 @@
+'use client';
+import { Dialog, DialogContent, DialogTitle } from '@/components/ui/dialog';
+import { getErrorMessage } from '@/util/util';
+import {
+  createConnectQueryKey,
+  useMutation,
+  useQuery,
+} from '@connectrpc/connect-query';
+import { GetAccountOnboardingConfigResponse } from '@neosync/sdk';
+import {
+  getAccountOnboardingConfig,
+  setAccountOnboardingConfig,
+} from '@neosync/sdk/connectquery';
+import { DialogDescription } from '@radix-ui/react-dialog';
+import { useQueryClient } from '@tanstack/react-query';
+import { ReactElement, useEffect, useState } from 'react';
+import { toast } from 'sonner';
+import { useAccount } from '../providers/account-provider';
+import { Skeleton } from '../ui/skeleton';
+import Configure from './Configure';
+import Connect from './Connect';
+import StepProgress from './StepProgress';
+import Sync from './Sync';
+import WelcomeOverview from './WelcomeOverview';
+import WelcomeRouter from './WelcomeRouter';
+
+export interface FormStep {
+  name: string;
+  component: JSX.Element;
+}
+
+export default function WelcomeDialog(): ReactElement {
+  const { account } = useAccount();
+  const {
+    data,
+    isLoading,
+    isFetching: isValidating,
+    error,
+  } = useQuery(
+    getAccountOnboardingConfig,
+    { accountId: account?.id ?? '' },
+    { enabled: !!account?.id }
+  );
+  const queryclient = useQueryClient();
+  const { mutateAsync: setOnboardingConfigAsync } = useMutation(
+    setAccountOnboardingConfig
+  );
+
+  const [currentStep, setCurrentStep] = useState<number>(0);
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+  const [showGuide, setShowGuide] = useState<boolean>(false);
+
+  const multiStepForm: FormStep[] = [
+    {
+      name: 'welcome',
+      component: (
+        <WelcomeOverview
+          currentStep={currentStep}
+          setCurrentStep={setCurrentStep}
+          setIsDialogOpen={setShowGuide}
+          completeForm={completeForm}
+        />
+      ),
+    },
+    {
+      name: 'connect',
+      component: (
+        <Connect currentStep={currentStep} setCurrentStep={setCurrentStep} />
```

Comment:
Most of these forms don't need to know their step or current step. They just need to know `onNext()` or `onBack()`, `onSkip()`, etc, whichever is relevant. Makes this a lot more readable and easier to manage.